pkg/ast/ast_operation_definition.go

```diff
@@ -16,6 +17,20 @@ const (
 	OperationTypeSubscription
 )
 
+func (t OperationType) Name() string {
+	switch t {
+	case OperationTypeUnknown:
```

Comment:
Why would someone use `OperationTypeUnknown` as OperationType?

Reply:
To initialize a variable with an invalid value or to return it e.g.

```go
op := OperationTypeUnknown
switch operationString {
    case "query":
    op = OperationTypeQuery
    ....
    default:
         return nil, fmt.Errorf("unknown operation %s", operationString)
```

Additionally, having an invalid value as the zero-value is also useful because if you forget to initialize a variable/field of `OperationType` it doesn't end up with a valid value.

Reply:
That would imply that `OperationTypeUnknown` is a valid value but it's not. There is only `query`, `mutation`, `subscription`. For your other arguments. You can always start with `query` and return an error if nothing matches.